deploy/configs/tools/modules.yaml

```diff
@@ -44,6 +44,8 @@ modules:
       - mvapich2
       - openmpi
       - osu-micro-benchmarks
+      - py-black
+      - py-cmake-format
```

Comment:
modules for these tools? I would avoid if possible. If necessary, may be we can wrap in python-dev?

Reply:
We could, but it gets a little intransparent. What's `python-dev` exactly? A common set of libraries to work with python, in which case `cmake-format` seems a little out of place?

Reply:
we created `python-dev` to wrap commonly used python packages together to avoid separate module for each package (otherwise we had to use `activate` for python extension).

it's true that `cmake-format` doesn't fit there. But it's not user facing package, right? In that case, can't you install package in $HOME or generate module?

Reply:
> we created `python-dev` to wrap commonly used python packages together to avoid separate module for each package (otherwise we had to use `activate` for python extension).

This was more meant as a rhetorical question. Just pointing out that at some point the name `python-dev` may become misleading. It's also a question if it is for the development of python or `python-tool`.

> it's true that `cmake-format` doesn't fit there. But it's not user facing package, right? In that case, can't you install package in $HOME or generate module?

I think in the long run it would be smart to run sever-side validation of code formatting instead of relying solely on hooks. Then it would be nice to have some modules for that.

Reply:
If we/you are using `spack diy`, installing package without exposing module would be ok right?

What I am trying to check is whether we can reduce the number of modules that we expose :)  Because this would be our next exercise (removing all old / un-needed spack+nix modules.

I am ok if you think its worth having the module.

Reply:
> If we/you are using `spack diy`, installing package without exposing module would be ok right?

I don't like having these formatting tools as explicit dependencies within Spack for software. It would be better if there would be a mixin for such things, but still…

From my point of view, it's better to check formatting at the CI level, too, and in that case modules come in pretty handy. Maybe we should just have a module for the HPC coding conventions itself (btw, is that going to be renamed à la `s/HPC//`)?

> What I am trying to check is whether we can reduce the number of modules that we expose :) Because this would be our next exercise (removing all old / un-needed spack+nix modules.

Sounds like a good idea.

> I am ok if you think its worth having the module.

You've got me convinced to remove the modules.
 
> P.S. I don't know what `py-black` module is for (as dependency).

Black is the Python code formatter without settings. I think @tristan0x mentioned that somewhere a while back and now his opinion has infiltrated me.